src/blueapi/service/interface.py

```diff
@@ -76,6 +80,35 @@ def stomp_client() -> StompClient | None:
         return None
 
 
+@cache
+def numtracker_client() -> NumtrackerClient | None:
+    conf = config()
+    if conf.numtracker is not None:
+        if conf.env.metadata is not None:
+            client = NumtrackerClient(url=conf.numtracker.url, headers={})
+            return client
+        else:
+            raise InvalidConfigError(
+                "Numtracker url has been configured, but there is no instrument or"
+                " instrument_session in the environment metadata"
+            )
+    else:
+        return None
+
+
+def _update_scan_num(md: dict[str, Any]) -> int:
+    numtracker = numtracker_client()
+    if numtracker is not None:
+        scan = numtracker.create_scan(md["instrument_session"], md["instrument"])
+        md["instrument_session_directory"] = str(scan.scan.directory.path)
+        return scan.scan.scan_number
+    else:
+        raise OSError(
```

Comment:
I think this should be a ValueError | InvalidConfigError not an OSError, there's nothing wrong with the OS.

Reply:
Done